base-compat.cabal

```diff
@@ -22,8 +23,12 @@ library
       -Wall
   build-depends:
       base == 4.*
-    , ghc-prim
     , setenv
+
+  if impl(ghc >= 7.2) && impl(ghc < 7.6)
+    -- Prior to GHC 7.6, GHC.Generics lived in ghc-prim
+    build-depends: ghc-prim
```

Comment:
Oh, I meant we could remove the condition for `ghc-prim`.  `base` already depends on `ghc-prim`, so always depending on it does not really add a dependency as it already is a transitive dependency of `base`.\

In general if I have two solutions that are functionally equivalent, I prefer the one with less branching, as for me it's easier to understand and reason about.